storage/paths/db_index.go

```diff
@@ -0,0 +1,891 @@
+
+	ftValid := false
+	for _, fileType := range storiface.PathTypes {
+		if fileType&ft == 0 {
+			ftValid = true
+			break
+		}
+	}
+	if !ftValid {
+		return xerrors.Errorf("Invalid filetype")
+	}
+
+	_, err := dbi.harmonyDB.Exec(ctx,
+		"DELETE FROM sectorlocation WHERE miner_id=$1 and sector_num=$2 and sector_filetype=$3 and storage_id=$4",
+		int(s.Miner), int(s.Number), int(ft), string(storageID))
+	if err != nil {
+		return xerrors.Errorf("StorageDropSector DELETE query fails: %v", err)
+	}
+
+	return nil
+}
+
+func (dbi *DBIndex) StorageFindSector(ctx context.Context, s abi.SectorID, ft storiface.SectorFileType, ssize abi.SectorSize, allowFetch bool) ([]storiface.SectorStorageInfo, error) {
+
+	var result []storiface.SectorStorageInfo
+
+	allowList := make(map[string]struct{})
+	storageWithSector := map[string]bool{}
+
+	type dbRes struct {
+		StorageId  string
+		Count      uint64
+		IsPrimary  bool
+		Urls       string
+		Weight     uint64
+		CanSeal    bool
+		CanStore   bool
+		Groups     string
+		AllowTo    string
+		AllowTypes string
+		DenyTypes  string
+	}
+
+	var rows []dbRes
+
+	fts := ft.AllSet()
+	// Find all storage info which already hold this sector + filetype
+	err := dbi.harmonyDB.Select(ctx, &rows,
+		`		SELECT DISTINCT ON (stor.storage_id)
+						  stor.storage_id,
+						  COUNT(*) OVER(PARTITION BY stor.storage_id) as count, 
+						  BOOL_OR(is_primary) OVER(PARTITION BY stor.storage_id) AS is_primary,
+						  urls,
+						  weight,
+						  can_seal,
+						  can_store,
+						  groups,
+						  allow_to,
+						  allow_types,
+						  deny_types
+						FROM sectorlocation sec
+						JOIN storagelocation stor ON sec.storage_id = stor.storage_id 
+						WHERE sec.miner_id = $1
+						  AND sec.sector_num = $2 
+						  AND sec.sector_filetype = ANY($3)  
+						ORDER BY stor.storage_id`,
+		s.Miner, s.Number, fts)
+	if err != nil {
+		return nil, xerrors.Errorf("Finding sector storage from DB fails with err: %v", err)
+	}
+
+	for _, row := range rows {
+
+		// Parse all urls
+		var urls, burls []string
+		for _, u := range splitString(row.Urls) {
+			rl, err := url.Parse(u)
+			if err != nil {
+				return nil, xerrors.Errorf("failed to parse url: %w", err)
+			}
+			rl.Path = gopath.Join(rl.Path, ft.String(), storiface.SectorName(s))
+			urls = append(urls, rl.String())
+			burls = append(burls, u)
+		}
+
+		result = append(result, storiface.SectorStorageInfo{
+			ID:         storiface.ID(row.StorageId),
+			URLs:       urls,
+			BaseURLs:   burls,
+			Weight:     row.Weight * row.Count,
+			CanSeal:    row.CanSeal,
+			CanStore:   row.CanStore,
+			Primary:    row.IsPrimary,
+			AllowTypes: splitString(row.AllowTypes),
+			DenyTypes:  splitString(row.DenyTypes),
+		})
+
+		storageWithSector[row.StorageId] = true
+
+		allowTo := splitString(row.AllowTo)
+		if allowList != nil && len(allowTo) > 0 {
```

Comment:
Shouldn't this be
```suggestion
		if allowTo != nil && len(allowTo) > 0 {
```

The current way won't ever set the allowList afaict

Reply:
This is exactly the logic in the index today: https://github.com/filecoin-project/lotus/blob/77594442471f6512d5657c5be9ca4511eea4f60b/storage/paths/index.go#L426

Keep in mind I've changed the name from allowTo (from the old index) to allowList( in the new index) to disambiguate allowTo from the DB